scripts/warnings-handling.py

```diff
@@ -0,0 +1,31 @@
+import os
+import re
+import sys
+
+
+warning_matcher = re.compile(r'.*warning[: ].*', re.I | re.S)
+warnings = 0
+
+def warnings_count():
+    global warnings
+    with open(os.path.join(os.getcwd(), 'build-log.txt'), 'r') as build_log:
+        build_log_lines = build_log.readlines()
+
+    for warning in build_log_lines:
+        if warning_matcher.match(warning) is not None:
+            print(warning_matcher.match(warning).group(0))
+            warnings += 1
+    return warnings
+
+def warnings_exit_code():
+    if warnings != 0:
+        print("Build has warnings.")
+        sys.exit(1)
```

Comment:
`sys.exit(1)` leads to the failed warning checking step. If compiling latest OpenCV does not generate any warnings, then it is fine. But since compiling OpenCV on macOS generates a lot of warnings, the warning checking step will always fail and may confuse both contributors and reviewers.

Reply:
Yes, the main idea was to fail a pipeline, if there is any warning. But I think this is not a good at the moment.